src/integrations/prefect-docker/tests/deployments/test_steps.py

```diff
@@ -201,20 +202,26 @@ def test_build_docker_image_raises_with_auto_and_existing_dockerfile():
         Path("Dockerfile").unlink()
 
 
-@pytest.mark.flaky(max_runs=3)
 def test_real_auto_dockerfile_build(docker_client_with_cleanup):
     os.chdir(str(Path(__file__).parent.parent / "test-project"))
+    image_name = "local/repo"
+    tag = f"test-{uuid4()}"
+    image_reference = f"{image_name}:{tag}"
     try:
         result = build_docker_image(
-            image_name="local/repo", tag="test", dockerfile="auto"
+            image_name=image_name, tag=tag, dockerfile="auto", pull=False
```

Comment:
The most important part here is `pull=False`, because we're using the prefect dev image that we built (not trying to pull an image from the public repo).  I think when we started installing `prefect` editable from the relative repo path, this would have started needing the locally built dev image